blacs/front_panel_settings.py

```diff
@@ -63,31 +66,40 @@ def restore(self):
         #            no: Restore as is
         #
         # Display errors, give option to cancel starting of BLACS so that the connection table can be edited
-        
+
         # Create saved connection table
         settings = {}
         question = {}
         error = {}
         tab_data = {'BLACS settings':{}}
         try:
+            with h5py.File(settings_path,'r') as h5file:
+                if 'connection table' in h5file:
+                    logger.info('Settings connection table found in file')
+        except:
+            logger.info('connection table not in file yet')
+            return settings,question,error,tab_data
```

Comment:
Admittedly, I haven't expressly tested this yet (I'll try that later today), but I don't think this works as intended. The check `if 'connection table' in h5file` shouldn't ever fail in an exception (it will just return `False` if it isn't present in the file). It isn't clear to me where the exception is being emitted from (partly because of the bare except). Is it actually coming from the fact the file doesn't exist?

I might suggest a simpler logic. If some exception is being emitted from somewhere, I'd prefer to catch it explicitly and log with a different message.
```python
with h5py.File(settings_path, 'r') as h5file:
    if 'connection table' not in h5file:
        logger.info('Front panel settings connection table not found!')
        return settings, question, error, tab_data
```

Reply:
No you're right, the try-except block won't catch a KeyError exception like intended since no ConnectionTable object is constructed yet, keeping it purely conditional should fix that

Reply:
After some testing I found out why the KeyError exception didn't appear again with the try-except logic:

`      
with h5py.File(settings_path,'r') as h5file:
`

uses "settings_path", which is a variable that doesn't exist (self.settings_path would have been correct) so we are moved to the except block, which should not have been bare. I've committed code with the simpler logic and a self.settings_path()

Reply:
I've also put down a TODO comment about the returns from the FronPanelSettings.restore() method since they'll be passed into the .restore_window() method, but this is only logged and does not produce an error window